python/ray/tests/test_reference_counting_2.py

```diff
@@ -699,6 +706,46 @@ def pass_nested_ref(borrower, outer_ref):
         time.sleep(1)
 
 
+def test_out_of_band_actor_handle_deserialization(shutdown_only):
+    ray.init(object_store_memory=100 * 1024 * 1024)
+
+    @ray.remote
+    class Actor:
+        def ping(self):
+            return 1
+
+    actor = Actor.remote()
+
+    @ray.remote
+    def func(config):
+        # deep copy will pickle and unpickle the actor handle.
+        config = copy.deepcopy(config)
+        return ray.get(config["actor"].ping.remote())
+
+    assert ray.get(func.remote({"actor": actor})) == 1
+
+
+def test_out_of_band_actor_handle_bypass_reference_counting(shutdown_only):
```

Comment:
Could we only have this test case? Or does the other one add additional value?

Reply:
Yea, the other one tests the borrowed case (that triggered the bug that this PR is fixing).